### ISSUE TITLE
`block-no-empty` false `needless-disable` in blocks with comments.

### ISSUE THREAD

**Issue by pamelalozano16** [stylelint/stylelint#7739]
### What minimal example or steps are needed to reproduce the bug?

```scss
/* stylelint-disable-next-line block-no-empty -- example. */
a {
  /* Comment */
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{ 
"rules": { "block-no-empty": true },
"reportNeedlessDisables": true
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYHkkwAeFxzaDEYA1kxHIwAtgAdcacuTSsaYggDo0KJAB0SNNMHWqsqNAGEIw4TDLK1JAL7qQAGhAAzIgQBycuIlnyYCjGAwMe3AIEmcAcwQQbRI0NFUQViwzKAwE+DQAbXU4uITsfEJSCkgwonDybBoSKBoAJygEnLQAXTtmhLqAVwI0hC0E7j4BITFcdLQsbpgrdtj4kDqYUQg6rFcYGChejAARanpeiamumHUrYNKIgDFV4RosKIArDFDg2FEgxBi8kAKCDhYCYJPAPGDYBJzX7-IpkchXcqVLDVWoNYEgUHJCEgc4gKxAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

No problems reported.

### What actually happened?

Error: `line 1, col 1 - Needless disable for "block-no-empty"`

**Comment by ybiquitous** [comment]
@pamelalozano16 Thanks for using the template and providing the reproducible demo.

This block seems to be empty since it includes a comment (`/* Comment */`). So the demo seems to work correctly.

For your reference, if specifying the `{"ignore": ["comments"]}` secondary option in the config, no lint errors happen. See this [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYHkkwAeFxzaDEYA1kxHIwAtgAdcacuTSsaYggDo0KJAB0SNNMHWqsqNAGEIw4TDLK1JAL7qQAGhAAzIgQBycuIlnyYCjGAwMe3AIEmcAcwQQbRI0NFUQViwzKAwE+DQAbXU4uITsfEJSCkgwonDybBoSKBoAJygEnLQAXTtmhLqAVwI0hC0E7j4BITFcdKysbpg7AZBykgg6mAnskJMzLD6Wqx322PiQZdElrFcYGChejAARanpeiamuletg0oiAMSXhGiwogBWGFCwVgoiCiBieRABQIHCwEwSeD+MGwCX20NhRTI5A+5UqWGqtQaiJAyOSaJA6isICsQA). Ref https://stylelint.io/user-guide/rules/block-no-empty#ignore-comments
